### PR TITLE
P4 tests update, remove dead code

### PR DESCRIPTION
**Pull request by senyai**
## Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/steps/source/p4.py [hunk 1]
```diff
@@ -108,10 +108,6 @@ def __init__(self, mode='incremental',
             config.error('p4branch should not end with a trailing / [p4branch = {}]'.format(
                     p4branch))
 
-        if (p4branch or p4extra_views) and not p4base:
```
Comment (p12tic) [inline master/buildbot/steps/source/p4.py:111]: It's not obvious to me that this is dead code. It seems that this case is not handled by the other conditionals around, is it?

Comment (senyai) [reply master/buildbot/steps/source/p4.py:111]: The logic is a bit obscure.
`not p4base` requires `p4viewspec`:
https://github.com/buildbot/buildbot/blob/913950c1f996c37e4aff1707e261af6a63b59db4/master/buildbot/steps/source/p4.py#L90
and `p4viewspec` requires `not (p4branch or p4extra_views)`:
https://github.com/buildbot/buildbot/blob/913950c1f996c37e4aff1707e261af6a63b59db4/master/buildbot/steps/source/p4.py#L93
I value code coverage more than having unreachable safety checks

Comment (p12tic) [reply master/buildbot/steps/source/p4.py:111]: Indeed, thanks for explanation.

**Comment by p12tic** [comment]
Looks good. Could you please fix too long lines which trigger flake8 and pylint?

I use this script to run the tests locally (adjust -j option to whatever parallelism you link):

```
. .venv/bin/activate
pylint -j64 --rcfile=common/pylintrc --spelling-private-dict-file=common/code_spelling_ignore_words.txt \
    master/buildbot master/docs/*.py master/setup.py \
    worker/buildbot_worker worker/setup.py \
    common/*.py
flake8 -j64 --config=common/flake8rc  \
    master/buildbot master/docs/*.py master/setup.py \
    worker/buildbot_worker worker/setup.py \
    common/*.py
```

`make pylint flake8` does the same, but they don't set the `-j` option, so are much slower.

**Comment by p12tic** [comment]
Thanks!

**Pull request merged**